ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java

```diff
@@ -132,30 +132,34 @@ private static void streamEndpointResponse(final Server server,
       final long startTimeNanos) {
     final WorkerExecutor workerExecutor = server.getWorkerExecutor();
     final VertxCompletableFuture<Void> vcf = new VertxCompletableFuture<>();
-    workerExecutor.executeBlocking(promise -> {
-      final OutputStream ros = new ResponseOutputStream(routingContext.response(),
-          streamingOutput.getWriteTimeoutMs());
-      routingContext.request().connection().closeHandler(v -> {
-        // Close the OutputStream on close of the HTTP connection
+    workerExecutor.executeBlocking(
+      promise -> {
+        final OutputStream ros = new ResponseOutputStream(routingContext.response(),
+                streamingOutput.getWriteTimeoutMs());
+        routingContext.request().connection().closeHandler(v -> {
+          // Close the OutputStream on close of the HTTP connection
+          try {
+            ros.close();
+          } catch (IOException e) {
+            promise.fail(e);
+          }
+        });
         try {
-          ros.close();
-        } catch (IOException e) {
+          streamingOutput.write(new BufferedOutputStream(ros));
+          promise.complete();
+        } catch (Exception e) {
           promise.fail(e);
+        } finally {
+          try {
+            ros.close();
+          } catch (IOException ignore) {
+            // Ignore - it might already be closed
+          }
         }
-      });
-      try {
-        streamingOutput.write(new BufferedOutputStream(ros));
-        promise.complete();
-      } catch (Exception e) {
-        promise.fail(e);
-      } finally {
-        try {
-          ros.close();
-        } catch (IOException ignore) {
-          // Ignore - it might already be closed
-        }
-      }
-    }, vcf);
+      },
+      false /*if this is true, worker execution blocks the main event loop*/,
```

Comment:
A lot of lines changed because I fixed the indentation, but this is what actually changed here.